api/api.go

```diff
@@ -193,75 +184,71 @@ func (b *BlockChainAPI) GetTransactionByBlockHashAndIndex(
 	ctx context.Context,
 	blockHash common.Hash,
 	index hexutil.Uint,
-) *RPCTransaction { // todo add error to the return type
+) (*RPCTransaction, error) { // todo add error to the return type
 	block, err := b.blocks.GetByID(blockHash)
 	if err != nil {
-		b.logger.Error().Err(err).Msg("failed to get block by id")
-		return nil
+		return handleError[*RPCTransaction](b.logger, err)
 	}
 
 	highestIndex := len(block.TransactionHashes) - 1
 	if index > hexutil.Uint(highestIndex) {
-		return nil
+		return nil, nil
```

Comment:
I guess we should return some kind of error here, instead of both values being `nil, nil`.